modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscriptionManager.java

```diff
@@ -274,6 +275,256 @@ private void removeLocalSubscriptionAndNotify(AndesSubscription subscription) th
+                        .equals(AMQPUtils.TOPIC_EXCHANGE_NAME)) {
+                    if (subscriptionIdPattern.equalsIgnoreCase(((DurableTopicSubscriber) subscription).getClientID())) {
+                        filteredSubscriptions.add(subscription);
+                    }
+                }
+            }else{
+                if (subscription.getSubscriptionId().toLowerCase().contains(subscriptionIdPattern.toLowerCase())) {
+                    filteredSubscriptions.add(subscription);
+                } else if (subscription.isDurable() && subscription.getStorageQueue().getMessageRouter().getName()
+                        .equals(AMQPUtils.TOPIC_EXCHANGE_NAME)) {
+                    if (((DurableTopicSubscriber) subscription).getClientID().toLowerCase().contains
+                            (subscriptionIdPattern.toLowerCase())) {
+                        filteredSubscriptions.add(subscription);
+                    }
+                }
+            }
+
+        }
+        return filteredSubscriptions;
+
+    }
+
+    /**
+     * Get inactive subscriptions filtered by identifier pattern and tokenized binding key
+     *
+     * @param isDurable true if searching for durable subscriptions
+     * @param protocolType protocol of the subscription
+     * @param destinationType type of subscription (QUEUE/TOPIC/DURABLE_TOPIC)
+     * @param bindingKeyPattern regex to match with binding key of subscriber
+     * @param subscriptionIdPattern regex to match with ID of the subscriber
+     * @param isExactMatchSubscriptionId exact match of subscription id or not
+     * @return Set of subscriptions filtered according to search criteria
+     * @throws AndesException
+     */
+    private Set<AndesSubscription> getInactiveSubscriptionsByTokenizedBindingKeyMatch(boolean isDurable,
+                                                                                     ProtocolType protocolType,
+                                                                                     DestinationType destinationType,
+                                                                                     String bindingKeyPattern,
+                                                                                     String subscriptionIdPattern,
+                                                                                     boolean isExactMatchSubscriptionId)
+            throws AndesException {
+
+        Set<AndesSubscription> filteredSubscriptions = new HashSet<>();
+        String messageRouter = destinationType.getAndesMessageRouter();
+
+        List<AndesSubscription> allInactiveSubscriptions = getInactiveSubscriberRepresentations();
+
+        for (AndesSubscription inactiveSubscription : allInactiveSubscriptions) {
+            if(isExactMatchSubscriptionId && inactiveSubscription.isDurable() == isDurable
+                    && inactiveSubscription.getStorageQueue().getMessageRouter().getName().equals(messageRouter)
```

Comment:
Again using a == on a boolean variable is a wrong use of the operator. We should able to simplify this by only using one of the variables here.